src/interface.jl

```diff
@@ -49,4 +49,6 @@ function createinstance(::Type{T}, args...) where {T}
     isconcretetype(T) ? bypass_constructor(T, args) : T(args...)
 end
 
-createinstance(::Type{T}, args...) where {T<:Tup} = T(args)
\ No newline at end of file
+createinstance(::Type{T}, args...) where {T<:Tup} = T(args)
+
+createinstance(::Type{T}) where {T} = (x...) -> createinstance(T, x...)
```

Comment:
```suggestion
struct Instantiator{T} end

Instantiator(::Type{T}) where {T} = Instantiator{T}()

(::Instantiator{T})(args...) where {T} = createinstance(T, args...)
```

This might be a little bit better, to avoid adding extra methods to an interface function (and in general I suspect it may be a little bit better for inference, compared with wrapping a type in a closure).

Reply:
Their's inference difference as `f(::Type{T}) where {T} = (x...) -> ...` would create a parametered function.
```julia
julia> f(::Type{T}) where {T} = (x...) -> T(1)
f (generic function with 1 method)

julia> typeof(f(Int))
var"#21#22"{Int64}
```
But yes we'd better not pollute the public interface.